components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java

```diff
@@ -293,24 +345,179 @@ public void testAddGetDeleteBlockConditions() throws Exception {
         Assert.assertEquals(blockConditionsUser.getConditionValue(), blockConditionsAddedUser.getConditionValue());
         Assert.assertEquals(blockConditionsIpRange.getStartingIP(), blockConditionsAddedIpRange.getStartingIP());
 
+        Assert.assertTrue(policyDAO.updateBlockConditionStateByUUID(uuidIp, true));
+        Assert.assertTrue(policyDAO.getBlockConditionByUUID(uuidIp).isEnabled());
+        Assert.assertTrue(policyDAO.getBlockConditions().size() == 5);
+
         Assert.assertTrue(policyDAO.deleteBlockConditionByUuid(uuidIp));
         Assert.assertTrue(policyDAO.deleteBlockConditionByUuid(uuidIpRange));
         Assert.assertTrue(policyDAO.deleteBlockConditionByUuid(uuidApi));
         Assert.assertTrue(policyDAO.deleteBlockConditionByUuid(uuidApp));
         Assert.assertTrue(policyDAO.deleteBlockConditionByUuid(uuidUser));
+
     }
 
     @Test
-    public void testAddGetDeleteCustomPolicy() throws Exception {
+    public void testAddGetUpdateDeleteCustomPolicy() throws Exception {
         PolicyDAO policyDAO = DAOFactory.getPolicyDAO();
         CustomPolicy customPolicy = SampleTestObjectCreator.createDefaultCustomPolicy();
         String uuid = policyDAO.addCustomPolicy(customPolicy);
 
         CustomPolicy policyAdded = policyDAO.getCustomPolicyByUuid(uuid);
         Assert.assertEquals(customPolicy.getSiddhiQuery(), policyAdded.getSiddhiQuery());
 
+        policyAdded.setDescription("updated custom policy");
+        policyDAO.updateCustomPolicy(policyAdded);
+        Assert.assertEquals(policyDAO.getCustomPolicyByUuid(uuid).getDescription() , "updated custom policy");
+
         policyDAO.deleteCustomPolicy(uuid);
         CustomPolicy policyDeletion = policyDAO.getCustomPolicyByUuid(uuid);
         Assert.assertNull(policyDeletion);
     }
+
+    @Test
+    public void testGetAllPolicies() throws Exception {
+        PolicyDAO policyDAO = DAOFactory.getPolicyDAO();
+        int size = policyDAO.getAllPolicies().size();
+        APIPolicy apiPolicy = SampleTestObjectCreator.createDefaultAPIPolicy();
+        ApplicationPolicy applicationPolicy = SampleTestObjectCreator.createDefaultApplicationPolicy();
+        SubscriptionPolicy subscriptionPolicy = SampleTestObjectCreator.createDefaultSubscriptionPolicy();
+        policyDAO.addApiPolicy(apiPolicy);
+        policyDAO.addApplicationPolicy(applicationPolicy);
+        policyDAO.addSubscriptionPolicy(subscriptionPolicy);
+        Set<PolicyValidationData> policyValidationData = policyDAO.getAllPolicies();
+        Assert.assertTrue(policyValidationData.size() == size + 3);
+    }
+
+
```

Comment:
Remove unwanted lines

Reply:
This is fixed with https://github.com/wso2/carbon-apimgt/pull/5147